skfda/preprocessing/registration/elastic.py

```diff
@@ -0,0 +1,780 @@
+    composition :math:`f^*(t)=f(\gamma^*(t))`.
+
+    If the template is not specified it is used the Karcher mean of the set of
+    functions under the elastic metric to perform the alignment, also known as
+    `elastic mean`, wich is the local minimum of the sum of squares of elastic
+    distances. See :func:`~elastic_mean`.
+
+    In [SK16-4-2]_ are described extensively the algorithms employed and
+    the SRSF framework.
+
+    Args:
+        template (str, :class:`FDataGrid` or callable, optional): Template to
+            align the curves. Can contain 1 sample to align all the curves to
+            it or the same number of samples than the fdatagrid. By default
+            `elastic mean`, in which case :func:`elastic_mean` is called.
+        penalty_term (float, optional): Controls the amount of elasticity.
+            Defaults to 0.
+        output_points (array_like, optional): Set of points where the
+            functions are evaluated, by default uses the sample points of the
+            fdatagrid which will be transformed.
+        grid_dim (int, optional): Dimension of the grid used in the DP
+            alignment algorithm. Defaults 7.
+        **kwargs: Named arguments to be passed to be passed to the callable
+            which constructs the template or to :func:`~elastic_mean` by
+            default.
+
+    Attributes:
+        template_ (:class:`FDataGrid`): Template learned during fitting,
+            used for alignment in :meth:`transform`.
+        warping_ (:class:`FDataGrid`): Warping applied during the last
+            transformation.
+
+    References:
+        ..  [SK16-4-2] Srivastava, Anuj & Klassen, Eric P. (2016). Functional
+            and shape data analysis. In *Functional Data and Elastic
+            Registration* (pp. 73-122). Springer.
+
+    Examples:
+
+        Elastic registration of with train/test sets.
+
+        >>> from skfda.preprocessing.registration import \
+        ...                                             ElasticRegistration
+        >>> from skfda.datasets import make_multimodal_samples
+        >>> X_train = make_multimodal_samples(n_samples=15, random_state=0)
+        >>> X_test = make_multimodal_samples(n_samples=3, random_state=1)
+
+        Fit the transformer, which learns the elastic mean of the train
+        set as template.
+
+        >>> elastic_registration = ElasticRegistration()
+        >>> elastic_registration.fit(X_train)
+        ElasticRegistration(...)
+
+        Registration of the test set.
+
+        >>> elastic_registration.transform(X_test)
+        FDataGrid(...)
+
+    """
+    def __init__(self, template="elastic mean", penalty=0., output_points=None,
+                 grid_dim=7, **kwargs):
+        """Initializes the registration transformer"""
+
+        self.template = template
+        self.penalty = penalty
+        self.output_points = output_points
+        self.grid_dim = grid_dim
+        self.kwargs = kwargs
```

Comment:
OK, so I'll delete it.

In some transformers I've seen that they have an argument called `**fit_params` in `fit`, maybe it could be useful to use it in this case.

Reply:
What kind of parameters are those you need to pass to the mean?

Reply:
Maybe to change the convergence parameters max_iter or tol, although the default params will produce a good alignment in general.

Reply:
For those a lambda or partial suffices (my answer was also for this conversation 😜).